odoo/addons/base/models/res_partner.py

```diff
@@ -753,6 +753,9 @@ def _get_name(self):
         return name
 
     def name_get(self):
+        # Prefetch the fields used by the `name_get`, so `browse` doesn't fetch other fields
+        self._origin._read(['is_company', 'name', 'parent_id', 'type', 'company_name', 'commercial_company_name'])
```

Comment:
You don't take `_context` keys into in account (used in `_name_get()`), which can lead to slow down when it used.

By example: when `show_email` is Truly in the context, you will fetch (`['is_company', 'name', 'parent_id', 'type', 'company_name', 'commercial_company_name']`) but not `email`. Then when the `email` will be access (`partner.email`), it will prefetch all fields of partner. In this case, it will less efficient than before (2 SQL request to fetch data instead 1, and some data fetch (your field in your _read) twice).

Actually there are also some context not used, maybe you can remove these deadcode.